src/preprocess_info.h

```diff
@@ -0,0 +1,34 @@
+#ifndef IE_NODE_PREPROCESS_INFO_H
+#define IE_NODE_PREPROCESS_INFO_H
+
+#include <napi.h>
+
+#include "inference_engine.hpp"
+
+namespace ienodejs {
+
+class PreProcessInfo : public Napi::ObjectWrap<PreProcessInfo> {
+ public:
+  static void Init(const Napi::Env& env);
+  static Napi::Object NewInstance(
+      const Napi::Env& envs,
+      InferenceEngine::PreProcessInfo& actual);
+  PreProcessInfo(const Napi::CallbackInfo& info);
+
+ private:
+  static Napi::FunctionReference constructor;
+  // APIs
+  void SetColorFormat(const Napi::CallbackInfo& info);
+  Napi::Value GetColorFormat(const Napi::CallbackInfo& info);
+  Napi::Value GetNumberOfChannels(const Napi::CallbackInfo& info);
+  void SetResizeAlgorithm(const Napi::CallbackInfo& info);
+  Napi::Value GetResizeAlgorithm(const Napi::CallbackInfo& info);
+  void SetVariant(const Napi::CallbackInfo& info);
+  Napi::Value GetMeanVariant(const Napi::CallbackInfo& info);
+
+  InferenceEngine::PreProcessInfo* actual_;
```

Comment:
Could we have a `InferenceEngine::InputInfo::Ptr _input_info;` here? With that, you can access the `PreProcessInfo` by `_input_info->getPreProcess()`.

Reply:
But I think this file will be a sub file of `input_info.h`

Reply:
What do you mean by "sub file"?

Reply:
I think there is a variable of class `InferenceEngine::InputInfo::Ptr` in the file `input_info.h`. If this file contains a variable of class `InferenceEngine::InputInfo::Ptr`, it is replicated. And this file, 'preprocess_info.h', which uses the variable of class `InferenceEngine::InputInfo::Ptr` plays similar role to the file `input_info.h`. I may can not express it accuractly, sorry,

Reply:
By using `InferenceEngine::InputInfo::Ptr` here, you can make sure the the instance of `InputInfo` will not be destructed when you want to use `PreProcessInfo`. Since `PreProcessInfo` is a member of `InputInfo`, if you only refer to `PreProcessInfo` by pointer, that pointer would be invalid if the hosting `InputInfo` gets destructed.

Reply:
Ok. I think the validity of the `PreProcessInfo` is more important. I will try to test now.